presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java

```diff
@@ -0,0 +1,242 @@
+                "User Memory",
+                "System Memory",
+                "Input Bytes",
+                "Output Bytes",
+                "Input Row Count",
+                "Output Row Count"));
+
+        Comparator<SqlTask> comparator = (a, b) -> {
+            TaskStats aTaskStats = a.getTaskInfo().getStats();
+            TaskStats bTaskStats = b.getTaskInfo().getStats();
+            return Long.compare(aTaskStats.getUserMemoryReservation().toBytes() + aTaskStats.getSystemMemoryReservation().toBytes(),
+                    bTaskStats.getUserMemoryReservation().toBytes() + bTaskStats.getSystemMemoryReservation().toBytes());
+        };
+        sqlTasks.stream()
+                .sorted(comparator.reversed())
+                .map(task -> {
+            TaskInfo taskInfo = task.getTaskInfo();
+            SqlTaskIoStats taskIOStats = task.getIoStats();
+            TaskStatus taskStatus = taskInfo.getTaskStatus();
+            TaskStats taskStats = taskInfo.getStats();
+            return ImmutableList.of(
+                    task.getQueryContext().getQueryId().toString(),
+                    task.getTaskId().toString(),
+                    taskStatus.getState().toString(),
+                    taskStats.getCreateTime().toString(),
+                    taskStats.getUserMemoryReservation().toString(),
+                    taskStats.getSystemMemoryReservation().toString(),
+                    Long.toString(taskIOStats.getInputDataSize().getTotalCount()),
+                    Long.toString(taskIOStats.getOutputDataSize().getTotalCount()),
+                    Long.toString(taskIOStats.getInputPositions().getTotalCount()),
+                    Long.toString(taskIOStats.getOutputPositions().getTotalCount()));
+        }).collect(toImmutableList());
+
+        logInfoTable(taskLogInfoList);
+    }
+
+    private static void logFormatted(ListMultimap<QueryId, SqlTask> queryIDToSqlTaskMap)
+    {
+        Comparator<Entry<QueryId, Long>> comparator = Comparator.comparingLong(Entry::getValue);
+        //Print summary of queries sorted from most memory hungry to least memory hungry
+        List<QueryId> queriesByMemoryUsage = queryIDToSqlTaskMap.asMap().entrySet().stream()
+                //Convert to Entry<QueryId, Long>, QueryId -> Total Memory Reservation
+                .map(entry ->
+                    new SimpleEntry<>(entry.getKey(), entry.getValue().stream()
+                            .map(SqlTask::getTaskInfo)
+                            .map(TaskInfo::getStats)
+                            .mapToLong(stats -> stats.getUserMemoryReservation().toBytes() + stats.getSystemMemoryReservation().toBytes())
+                            .sum())
+                ).sorted(comparator.reversed())
+                .map(Entry::getKey)
+                .collect(toImmutableList());
+        queriesByMemoryUsage.stream()
+                .forEach(queryId -> logFormatted(queryId, queryIDToSqlTaskMap.get(queryId)));
```

Comment:
do we want to have one table per query? Or should it be a single table that contains multiple queries?